### PR TITLE
MDEXP-237: Omit calling SRS for custom Profile

### PR DESCRIPTION
**Pull request by KVupp**
https://issues.folio.org/browse/MDEXP-237

## Purpose
- For Q3 -2020 for records generated from custom profile we are not fetching records from SRS, and always need to generate on the fly

## Approach
- Change to check if custom mapping profile and call only inventory in that case
- Just disabled the old tests with comments
- Added a new method to generate all the holdings and items json mapping
- Corrected mapping path for call number fields
- Added new tests that test all the fields for holdings

ADDITIONAL
- Removed duplicate values from ReferenceDataImpl. Left material types in there because it uses different key. May need to clean that up later

#### TODOS and Open Questions
- [ ] Possibly test this on rancher if all the holding mappings are being generated correctly

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/service/transformationfields/TransformationConfigConstants.java [hunk 1]
```diff
@@ -45,7 +45,7 @@
     ELECTRONIC_ACCESS_LINKTEXT_DEFAULT,
     ELECTRONIC_ACCESS_MATERIALS_SPECIFIED_DEFAULT,
     ELECTRONIC_ACCESS_PUBLICNOTE_DEFAULT,
-    ID,
+    HOLDINGS_UUID,
```
Comment (andrii-novytskyi) [inline src/main/java/org/folio/service/transformationfields/TransformationConfigConstants.java:48]: ID is a common field for all types, why we specified it for holdings?

**Comment by andrii-novytskyi** [comment]
Don`t we need to wide MappingServiceUnitTest with all fields for holdings in the scope of this ticket?

**Review by illia-borysenko — APPROVED** [review]

**Comment by KVupp** [comment]
> Don`t we need to wide MappingServiceUnitTest with all fields for holdings in the scope of this ticket?

Good point Andrii , I added that test. On the way also changed a few things , added those in the description:)

File: src/main/java/org/folio/service/transformationfields/TransformationConfigConstants.java (hunk 1 above)
Comment (KVupp) [reply src/main/java/org/folio/service/transformationfields/TransformationConfigConstants.java]: yup that was a left over cleanup, changed it now. Thanks !

**Review by andrii-novytskyi — APPROVED** [review]

**Pull request merged**